dds/tests/listeners.rs

```diff
@@ -2236,15 +2136,7 @@ fn data_writer_offered_incompatible_qos_listener() {
         )
         .unwrap();
 
-    let cond = writer.get_statuscondition().unwrap();
-    cond.set_enabled_statuses(&[StatusKind::OfferedIncompatibleQos])
-        .unwrap();
-
-    let mut wait_set = WaitSet::new();
-    wait_set
-        .attach_condition(Condition::StatusCondition(cond))
-        .unwrap();
-    wait_set.wait(Duration::new(10, 0)).unwrap();
+    std::thread::sleep(std::time::Duration::from_secs(1));

```

Comment:
I think, the too early triggering needs to be fixed first.

Reply:
I am quite sure the behavior itself is correct and that it used to work only because of the poor performance. If the listener is triggered before calling wait then the wait set should only be triggered if the event it is waiting for occurs again. What we could do though is to have a different blocking mechanism in the listener to release the test thread instead of using and relying on the mock. That would likely make the test clearer and focus on more on the point under test which is actually the listener itself

Reply:
The listener test are now refactored to actually test only the listener. This is done by using an std channel to send a message from the listener to the test thread and comparing the expected results. This modification also removed the usage of the mocks so the mockall crate dev dependency is removed.